backend/database/migrations/2024_11_16_221022_add_room_code_field.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('rooms', function (Blueprint $table) {
+            $table->integer('roomcode')->unique()->nullable();
```

Comment:
i think the number as a string would be more flexible and easier to handle, especially if we need to change the format in the future

another point: does unique add an index automatically? because if it doesn't we should add an index as we'll be searching for it a lot